index.d.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: strong-error-handler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// Type definitions for strong-error-handler 3.x
+// Project: https://github.com/strongloop/strong-error-handler
+// Definitions by: Kyusung Shim <https://github.com/shimks>
+// TypeScript Version: 2.9
+
+import * as Express from 'express';
+
+export = errorHandlerFactory;
+
+/**
+ * Creates a middleware function for error-handling
+ * @param options Options for error handler settings
+ */
+declare function errorHandlerFactory(
+  options?: errorHandlerFactory.ErrorHandlerOptions
+): errorHandlerFactory.StrongErrorHandler;
+
+declare namespace errorHandlerFactory {
+
+  /**
+   * Error-handling middleware function. Includes server-side logging
+   */
+  type StrongErrorHandler = (
+    err: Error,
+    req: Express.Request,
+    res: Express.Response,
+    next: (err?: any) => void
+  ) => void;
+
+  /**
+   * Writes thrown error to response
+   * @param err Error to handle
+   * @param req Incoming request
+   * @param res Response
+   * @param options Options for error handler settings
+   */
+  function writeErrorToResponse(
+    err: Error,
+    req: Express.Request,
+    res: Express.Response,
+    options?: errorHandlerFactory.ErrorHandlerOptions
+  ): void;
+
+  /**
+   * Options for error-handling
+   */
+  interface ErrorHandlerOptions {
+    debug?: boolean;
+    log?: boolean;
```

Comment:
Should we split this interface into two smaller interfaces, one containing the options used by both the middleware and the writer, the other adding options specific to the middleware (i.e. `log`)?

```ts
interface ErrorHandlerCoreOptions {
  debug?: boolean;
  // etc.
}

interface ErrorHandlerMiddlewareOptions extends ErrorHandlerCoreOptions {
  log?: boolean;
}
```

Reply:
Well, the `log` property is not supported by `writeErrorToResponse`. Ideally, the interface used by `writeErrorToResponse` options should not offer properties that are not supported by the function.

```ts
declare function errorHandlerFactory(
 options?: errorHandlerFactory.ErrorHandlerMiddlewareOptions
: errorHandlerFactory.StrongErrorHandler;

function writeErrorToResponse(
  err: Error,
  req: Express.Request,
  res: Express.Response,
  options?: errorHandlerFactory.ErrorHandlerCoreOptions
): void;
```

On the second thought, I think the following interface names may be better because they are shorter:
 - `ErrorHandlerOptions` (used by middleware factory)
 - `ErrorWriterOptions` (used by writeErrorToResponse)

Anyhow, this is not a big deal, I am ok with the current version where there is a single interface only.